torchao/quantization/linear_activation_static_quantized_tensor.py

```diff
@@ -0,0 +1,193 @@
+        aqt = input_quant_func_static(
+            input_tensor, scale=scale, zero_point=zero_point, **quant_kwargs
+        )
+        return torch.nn.functional.linear(aqt, original_weight_tensor, bias)
+
+    @classmethod
+    def from_float(
+        cls,
+        input_float: torch.Tensor,
+        input_quant_func: Callable,
+        scale: torch.Tensor,
+        zero_point: Optional[torch.Tensor] = None,
+        quant_kwargs: Dict[str, Any] = {},
+    ):
+        return cls(input_float, input_quant_func, scale, zero_point, quant_kwargs)
+
+    def _apply_fn_to_data(self, fn):
+        return self.__class__(
+            fn(self.original_weight_tensor),
+            self.input_quant_func_static,
+            fn(self.scale),
+            fn(self.zero_point) if self.zero_point is not None else None,
+            self.quant_kwargs,
+        )
+
+    def to(self, *args, **kwargs):
+        kwargs = self._get_to_kwargs(*args, **kwargs)
+        device = kwargs.pop("device")
+        return self.__class__(
+            self.original_weight_tensor.to(**kwargs),
+            self.input_quant_func_static,
+            self.scale.to(device),
+            self.zero_point.to(device) if self.zero_point is not None else None,
+            self.quant_kwargs,
+        )
+
+
+implements = LinearActivationQuantizedTensorStatic.implements
+
+
+@implements(torch.nn.functional.linear)
+def _(func, types, args, kwargs):
+    input_tensor, weight_tensor, bias = (
+        args[0],
+        args[1],
+        args[2] if len(args) > 2 else None,
+    )
+    if isinstance(weight_tensor, LinearActivationQuantizedTensorStatic):
+        return weight_tensor._quantized_linear_op(input_tensor, weight_tensor, bias)
+
+    raise NotImplementedError(
+        "LinearActivationQuantizedTensor: No specialized dispatch found for linear op"
+    )
+
+
+@implements(aten.detach.default)
+def _(func, types, args, kwargs):
+    return return_and_correct_aliasing(
+        func, args, kwargs, args[0]._apply_fn_to_data(torch.detach)
+    )
+
+
+@implements(aten.clone.default)
+def _(func, types, args, kwargs):
+    return return_and_correct_aliasing(
+        func, args, kwargs, args[0]._apply_fn_to_data(torch.clone)
+    )
+
+
+@implements(aten._to_copy.default)
+def _(func, types, args, kwargs):
+    return return_and_correct_aliasing(
+        func,
+        args,
+        kwargs,
+        args[0].to(*args[1:], **kwargs)._apply_fn_to_data(torch.clone),
+    )
+
+
+@implements(aten.t.default)
+def _(func, types, args, kwargs):
+    return return_and_correct_aliasing(
+        func, args, kwargs, args[0]._apply_fn_to_data(torch.t)
+    )
+
+
+to_linear_activation_quantized_static = LinearActivationQuantizedTensorStatic.from_float
+
+if TORCH_VERSION_AT_LEAST_2_5:
+    # Allow a model with LinearActivationQuantizedTensor weights to be loaded with `weights_only=True`
+    torch.serialization.add_safe_globals([LinearActivationQuantizedTensorStatic])
```

Comment:
nit: maybe rename it to `LinearActivationStaticQuantizedTensor` to align with the file name